### ISSUE TITLE
where is the test code? how to test images? could you please help me?

### ISSUE THREAD

**Issue by kashifibnaejaz** [vincrichard/EfficientPS#3]

**Comment by vincrichard** [comment]
There is no test code in itself, at each epoch the model create and save all the validation images. At the end of the training you can use the weight of the model and load it in your own code to infer new images.

**Comment by kashifibnaejaz** [comment]
thanks a lot

**Comment by kashifibnaejaz** [comment]
Hey, thanks a lot for helping me last time
please could you explain in which part of the code you are  producing instances 
IN INSTANCE HEAD
 def forward(self, inputs, targets={}):
     27         losses = {}
---> 28         proposals, losses_rpn = self.rpn(inputs, targets['instance'])
     29         if self.training:
     30             _, losses_head = self.roi_heads(inputs, proposals, targets['instance'])
it's showing error when i test the image. if i am not wrong, are the annotations instances?
in which part of the code model produces instance images


-

**Comment by vincrichard** [comment]
Hello,
Do you mean the target instances ? 
This is in the panoptic_dataset.py file, starting from [here](https://github.com/vincrichard/EfficientPS/blob/e8d0548a05f816293a0adb596ec177f174335010/datasets/panoptic_dataset.py#L104) I am extracting the instance information from the dataset and creating an `Instance` object which is base on Detectron2. 

After I don't remember quite well the processing I made but I extracted the feature from the dataset to create this object. which is stored in target['instance']

**Comment by nmddc0211** [comment]
After looking at the code for some time, I think the features that you extracted for the Instance object (based on detectron2) are binary masks, the corresponding class and the bbox. The Instance object holds not only one instance but all the instances of the thing classes of one image. 
        # Create same size of bbox and mask instance
        if len(rpn_bbox) > 0:
            rpn_bbox = coco_to_pascal_bbox(np.stack([*rpn_bbox]))

            instance.gt_masks = BitMasks(np.array(instance_mask))
            instance.gt_classes = torch.as_tensor(instance_cls)
            instance.gt_boxes = Boxes(rpn_bbox)
        else:
            instance.gt_masks = BitMasks(torch.Tensor([]).view(0, 1, 1))
            instance.gt_classes = torch.as_tensor([])
            instance.gt_boxes = Boxes([])
instance_mask, instance_cls, rpn_bbox are basically the lists which contain the features of the thing instances. They have the same length (which is the number of thing instances) and the same index corresponds to the same one instance.

**Comment by kashifibnaejaz** [comment]
thanks a lot
when i have to test the image i have to pass the instance(instance annotation of the image)
the model is not producing the instances in itself. What you refered last time , u are passing the instances from validation.json and the model is not producing segmented masks in itself
